### PR TITLE
Support AirBnB's Smartstack Nerve client for SD

### PR DESCRIPTION
**Pull request by PrFalken**
Following our discussion on Dev mailing list :
Nerve's registration format differs from serverset. With this commit
there is now a dedicated zookeeper file for the base ZK watch support,
and two separate files for serverset and nerve.

Reference:
https://github.com/airbnb/nerve

### CODE REVIEW

File: retrieval/discovery/nerve.go [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+}
+
+func parseNerveMember(data []byte, path string) (*model.LabelSet, error) {
+	member := nerveMember{}
+	err := json.Unmarshal(data, &member)
+	if err != nil {
+		return nil, fmt.Errorf("error unmarshaling nerve member %q: %s", path, err)
+	}
+
+	labels := model.LabelSet{}
```
Comment (brian-brazil) [inline retrieval/discovery/nerve.go]: You should put the name in a label too

File: retrieval/discovery/zookeeper.go [hunk 2]
```diff
@@ -0,0 +1,213 @@
+package discovery
```
Comment (brian-brazil) [inline retrieval/discovery/zookeeper.go]: I meant to put this under the `util` directory. This is the only Go implementation of a Zookeeper TreeCache, and at least one person was looking to use it elsewhere.

Comment (PrFalken) [inline retrieval/discovery/zookeeper.go]: OK I thought I would do this in a second PR.
How would you like to name the package, as we already import "zk" ?

Comment (brian-brazil) [inline retrieval/discovery/zookeeper.go]: treecache is probably best

**Comment by brian-brazil** [comment]
That looks good, can you add some config for it to `conf.good.yml` to protect against regressions and squash your commits?

**Comment by PrFalken** [comment]
Hello Brian, config testing and squashes done.

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**